cicecore/cicedynB/dynamics/ice_dyn_shared.F90

```diff
@@ -1564,6 +1564,111 @@ subroutine strain_rates_T (nx_block,   ny_block, &
+                                 uvelE,      vvelE,     &
+                                 uvelN,      vvelN,     &
+                                 uvelU,      vvelU,     &
+                                 dxE,        dyN,       &
+                                 dxU,        dyU,       &
+                                 ratiodxN,   ratiodxNr, &
+                                 ratiodyE,   ratiodyEr, &
+                                 epm,  npm,  uvm,       &
+                                 divU,       tensionU,  &
+                                 shearU,     DeltaU     )
+
+      integer (kind=int_kind), intent(in) :: &
+         nx_block, ny_block    ! block dimensions
+         
+      integer (kind=int_kind) :: &
+         i, j                  ! indices
+         
+      real (kind=dbl_kind), dimension (nx_block,ny_block), intent(in) :: &
+         uvelE    , & ! x-component of velocity (m/s) at the E point
+         vvelE    , & ! y-component of velocity (m/s) at the N point
+         uvelN    , & ! x-component of velocity (m/s) at the E point
+         vvelN    , & ! y-component of velocity (m/s) at the N point
+         uvelU    , & ! x-component of velocity (m/s) interp. at U point
+         vvelU    , & ! y-component of velocity (m/s) interp. at U point
+         dxE      , & ! width of E-cell through the middle (m)
+         dyN      , & ! height of N-cell through the middle (m)
+         dxU      , & ! width of U-cell through the middle (m)
+         dyU      , & ! height of U-cell through the middle (m)
+         ratiodxN , & ! -dxN(i+1,j)/dxN(i,j) for BCs
+         ratiodxNr, & ! -dxN(i,j)/dxN(i+1,j) for BCs
+         ratiodyE , & ! -dyE(i,j+1)/dyE(i,j) for BCs
+         ratiodyEr, & ! -dyE(i,j)/dyE(i,j+1) for BCs
+         epm      , & ! E-cell mask
+         npm      , & ! E-cell mask
+         uvm          ! U-cell mask
+         
+         
+      real (kind=dbl_kind), intent(out):: &
+        divU, tensionU, shearU, DeltaU      ! strain rates at the T point
+
+      ! local variables
+
+      real (kind=dbl_kind) :: &                     
+        uNip1j, uNij, vEijp1, vEij, uEijp1, uEij, vNip1j, vNij
+      
+      character(len=*), parameter :: subname = '(strain_rates_U)'
+         
+      !-----------------------------------------------------------------
+      ! strain rates
+      ! NOTE these are actually strain rates * area  (m^2/s)
+      !-----------------------------------------------------------------
+
+      uNip1j = uvelN(i+1,j) * npm(i+1,j) &
+             +(npm(i,j)-npm(i+1,j)) * npm(i,j) * ratiodxN(i,j) * uvelN(i,j)
+      uNij   = uvelN(i,j) * npm(i,j) &
+             +(npm(i+1,j)-npm(i,j)) * npm(i+1,j) * ratiodxNr(i,j) * uvelN(i+1,j)
+      vEijp1 = vvelE(i,j+1) * epm(i,j+1) &
+             +(epm(i,j)-epm(i,j+1)) * epm(i,j) * ratiodyE(i,j) * vvelE(i,j)
+      vEij   = vvelE(i,j) * epm(i,j) &
+             +(epm(i,j+1)-epm(i,j)) * epm(i,j+1) * ratiodyEr(i,j) * vvelE(i,j+1)
```

Comment:
This would read ore clearly if the different  `ratiod**` term were aligned, I think. We could also align the terms of the  substractions (on the continuation lines).

Reply:
I agree.  We've all decided to try do a better job of aligning "similar" multi-line equations.  This is really, really helpful with things like these equations where it's so helpful to see how the indexing is working.  I would align the parens and commas and other things as much as possible.  A lot (not all) of the current solver implementation does this.  We should do the same for the new implementation.  It does really help readability and reviewability.

Reply:
It would be good to look over some of the new implementation already PR'ed, some of those could also be cleaned up a bit to make them more readable.

Also, when I say "we've all decided", that was a discussion we had on this side of the code camp, we hope to pull you along :)